homeassistant/components/flux_led/light.py

```diff
@@ -308,101 +422,95 @@ def effect(self):
+
+    @property
+    def device_info(self):
+        """Return the device information."""
+        device_name = "FluxLED/Magic Home"
+        device_model = "LED Lights"
+
+        return {
+            ATTR_IDENTIFIERS: {(DOMAIN, self._unique_id)},
+            ATTR_NAME: self._name,
+            ATTR_MANUFACTURER: device_name,
+            ATTR_MODEL: device_model,
+        }
 
+    async def async_turn_on(self, **kwargs):
+        """Turn on the light."""
+
+        rgb = None
         hs_color = kwargs.get(ATTR_HS_COLOR)
 
         if hs_color:
             rgb = color_util.color_hs_to_RGB(*hs_color)
-        else:
-            rgb = None
 
         brightness = kwargs.get(ATTR_BRIGHTNESS)
         effect = kwargs.get(ATTR_EFFECT)
         white = kwargs.get(ATTR_WHITE_VALUE)
-        color_temp = kwargs.get(ATTR_COLOR_TEMP)
-
-        # handle special modes
-        if color_temp is not None:
-            if brightness is None:
-                brightness = self.brightness
-            if color_temp > COLOR_TEMP_WARM_VS_COLD_WHITE_CUT_OFF:
-                self._bulb.setRgbw(w=brightness)
-            else:
-                self._bulb.setRgbw(w2=brightness)
-            return
 
-        # Show warning if effect set with rgb, brightness, or white level
-        if effect and (brightness or white or rgb):
-            _LOGGER.warning(
-                "RGB, brightness and white level are ignored when"
-                " an effect is specified for a flux bulb"
+        if effect == EFFECT_RANDOM:
+            color_red = random.randint(0, 255)
+            color_green = random.randint(0, 255)
+            color_blue = random.randint(0, 255)
+
+            self._bulb.setRgbw(
+                r=color_red,
+                g=color_green,
+                b=color_blue,
             )
 
-        # Random color effect
-        if effect == EFFECT_RANDOM:
-            self._bulb.setRgb(
-                random.randint(0, 255), random.randint(0, 255), random.randint(0, 255)
+            self._hs_color = color_util.color_RGB_to_hs(
+                color_red,
+                color_green,
+                color_blue,
             )
-            return
+            self._last_update = time.time()
 
-        if effect == EFFECT_CUSTOM:
-            if self._custom_effect:
-                self._bulb.setCustomPattern(
-                    self._custom_effect[CONF_COLORS],
-                    self._custom_effect[CONF_SPEED_PCT],
-                    self._custom_effect[CONF_TRANSITION],
-                )
             return
 
-        # Effect selection
         if effect in EFFECT_MAP:
             self._bulb.setPresetPattern(EFFECT_MAP[effect], self._effect_speed)
-            return
 
-        # Preserve current brightness on color/white level change
-        if brightness is None:
-            brightness = self.brightness
+        if not brightness:
+            brightness = self._last_brightness
+
+        self._last_brightness = brightness
```

Comment:
Looks like we don't need `_last_brightness` etc but can use `_brightness` etc instead.

Reply:
Unfortunately the LED lights actually return a color and brightness value even when off which may not reflect the last value. I have updated/simplified now though to only capture the last brightness and color when turning the light off so it is used properly when the light is turned back on.